File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        ans = [0] * len(temperatures)
        stack = []
        for i, temp in enumerate(temperatures):
            while stack and temp > stack[-1][0]:
                cur = stack.pop() # pops from back of list
                ans[cur[1]] = i - cur[1]
            stack.append([temp, i])
        return ans

File: Data Structures & Algorithms/reverse-string/submission-0.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        stack<char> word;
        for (char c : s) word.push(c);
        int i{0};
        for (int i{0}; i < s.size(); i++) {
            s[i] = word.top();
            word.pop();
        }
    }
};